pkgs/development/python-modules/sqlalchemy/default.nix

```diff
@@ -6,32 +6,22 @@
 , mock
 , isPy3k
 , pysqlite
+, pytest_xdist
 }:
 
 buildPythonPackage rec {
   pname = "SQLAlchemy";
-  version = "1.2.14";
+  version = "1.3.5";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "9de7c7dabcf06319becdb7e15099c44e5e34ba7062f9ba10bc00e562f5db3d04";
+    sha256 = "0bsc11cdvy0q3vkrl106n5i5gxciyzd2axdxir2l6m7r1bb2a2f3";
   };
 
-  patches = [
-    # fix for failing doc tests
-    # https://bitbucket.org/zzzeek/sqlalchemy/issues/4370/sqlite-325x-docs-tutorialrst-doctests-fail
-    (fetchpatch {
-      name = "doc-test-fixes.patch";
-      url = https://bitbucket.org/zzzeek/sqlalchemy/commits/63279a69e2b9277df5e97ace161fa3a1bb4f29cd/raw;
-      sha256 = "1x25aj5hqmgjdak4hllya0rf0srr937k1hwaxb24i9ban607hjri";
-    })
-  ];
-
   checkInputs = [
     pytest
     mock
-#     Disable pytest_xdist tests for now, because our version seems to be too new.
-#     pytest_xdist
+    pytest_xdist
```

Comment:
this plugin is of no use for us so do not include it

Reply:
I tried not adding this, and the `checkPhase` fails with the following message:

```
ERROR: usage: py.test [options] [file_or_dir] [file_or_dir] [...]
py.test: error: unrecognized arguments: --max-worker-restart=5
  inifile: /build/SQLAlchemy-1.3.5/setup.cfg
  rootdir: /build/SQLAlchemy-1.3.5

builder for '/nix/store/4zac899pfwph14f01ycsylf9n1rgwg2g-python3.7-SQLAlchemy-1.3.5.drv' failed with exit code 4
error: build of '/nix/store/4zac899pfwph14f01ycsylf9n1rgwg2g-python3.7-SQLAlchemy-1.3.5.drv' on 'ssh://builder@13.56.251.157' failed: builder for '/nix/store/4zac899pfwph14f01ycsylf9n1rgwg2g-python3.7-SQLAlchemy-1.3.5.drv' failed with exit code 4
builder for '/nix/store/4zac899pfwph14f01ycsylf9n1rgwg2g-python3.7-SQLAlchemy-1.3.5.drv' failed with exit code 1
error: build of '/nix/store/4zac899pfwph14f01ycsylf9n1rgwg2g-python3.7-SQLAlchemy-1.3.5.drv', '/nix/store/rxarls8spq805ra976bf23glwwv804q8-python3.7-snowflake-sqlalchemy-1.1.13.drv' failed
```

Reply:
you can also do a default pytest run, for most packages this should be all you need:

```
checkPhase = ''
  pytest
''';
```

Reply:
Trying it, looks promising so far.

Reply:
Never mind, it's actually not working for me. I tried different combinations and I can't get it to work without `pytest_xdist`.

Using `pytest` I get

```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --max-worker-restart=5
  inifile: /build/SQLAlchemy-1.3.5/setup.cfg
  rootdir: /build/SQLAlchemy-1.3.5
```

Using `pytest test`, I get

```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --max-worker-restart=5
  inifile: /build/SQLAlchemy-1.3.5/setup.cfg
  rootdir: /build/SQLAlchemy-1.3.5
```

That's coming from [here](https://github.com/sqlalchemy/sqlalchemy/blob/190e0139e834e4271268652e058c280787ae69eb/setup.cfg#L8). Is this worth a patch or is there a way to fix it? I don't write Python, I'm not familiar with these tools.

Reply:
always the nuke from orbit option, and remove that entire line
```
postInstall = ''
  sed '/max-worker-restart/d' setup.cfg > setup.cfg
'';

Reply:
I managed to fix with the following commit:

```patch
commit adae23a9b0fc6b146328a1f4e97dd4ee19e35d1f
Author: Wael M. Nasreddine <wael.nasreddine@gmail.com>
Date:   Fri Jul 19 17:01:32 2019 -0700

    pythonPackages.sqlalchemy: disable pytest_xdist and fix the tests

diff --git a/pkgs/development/python-modules/sqlalchemy/default.nix b/pkgs/development/python-modules/sqlalchemy/default.nix
index 6a5075154fa..e67ae99560d 100644
--- a/pkgs/development/python-modules/sqlalchemy/default.nix
+++ b/pkgs/development/python-modules/sqlalchemy/default.nix
@@ -6,7 +6,6 @@
 , mock
 , isPy3k
 , pysqlite
-, pytest_xdist
 }:
 
 buildPythonPackage rec {
@@ -21,11 +20,14 @@ buildPythonPackage rec {
   checkInputs = [
     pytest
     mock
-    pytest_xdist
   ] ++ lib.optional (!isPy3k) pysqlite;
 
+  postInstall = ''
+    sed -e 's:--max-worker-restart=5::g' -i setup.cfg
+  '';
+
   checkPhase = ''
-    py.test -k "not test_round_trip_direct_type_affinity"
+    pytest test
   '';
 
   meta = with lib; {

```

@FRidh is this worth it for removing pytest_xdist?

Reply:
you may want to be careful with sed with -i, I believe only the GNU versions have it.

Reply:
> you may want to be careful with sed with -i, I believe only the GNU versions have it.

Nix only has the GNU version anyway, unless you know something I do not.

Reply:
I'm unfamiliar with what gets packaged as part of coreutils, if there's the guarantee that it will always be GNU, great :)